### PR TITLE
Updated CardModal text input to textarea so it looks better for longer card titles

### LINKED ISSUE #76: Fix the text input when clicking on a card modal with long content

### PR DESCRIPTION
**Pull request by rizwan171**
resolves #76

### CODE REVIEW

File: src/components/CardModal/CardModal.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const CardModal = ({ card }) => {
   const [moveOpen, setMoveOpen] = useState(false);
   const [selectedListId, setSelectedListId] = useState("");
   const [editableContent, setEditableContent] = useState(card.content);
-  const [width, setWidth] = useState(20);
+  const [rows, setRows] = useState(1);
```
Comment (rizwan171) [inline src/components/CardModal/CardModal.js:16]: The input element makes use of width, but for the text area, I've just set this to w-full below.
This state has been changed to control the amount of rows a text area has instead.
Rows correspond (roughly) to the height of a textarea based on sentences i.e.
If rows is 1, the height is set such that only one sentence is shown and you have to scroll to see the rest

File: src/components/CardModal/CardModal.js [hunk 2]
```diff
@@ -22,10 +22,10 @@ const CardModal = ({ card }) => {
...
     setEditableContent(e.target.value);
 
-    const inputLength = e.target.value.length;
-    if (inputLength >= 20 && inputLength < 40) {
-      setWidth(e.target.value.length);
-    }
+    // The scrollHeight is the height of all the content, including that which exceeds 1 row of textarea.
+    // clientHeight is the height of the content the user can see at this point with 1 row of textarea.
+    // To dynamically set the rows, we simply divide the scrollHeight by clientHeight and roundup
+    setRows(Math.ceil(e.target.scrollHeight / e.target.clientHeight));
```
Comment (rizwan171) [inline src/components/CardModal/CardModal.js:28]: Rows is set dynamically based on the card title. We always round up to make sure it covers hanging words or letters

File: src/components/CardModal/CardModal.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ const CardModal = ({ card }) => {
   const handleOnBlur = () => {
     setSelected(false);
     setEditableContent(card.content);
+    setRows(1);
```
Comment (rizwan171) [inline src/components/CardModal/CardModal.js:42]: When we close the edit title textarea, we want to reset the rows as they'll be calculated again on focus. If we do not set this, the calculations in `handleOnFocus` below will be incorrect

**Comment by rizwan171** [comment]
**Steps to test**
1. Pull this branch, load this app, create a card with a paragraph as its content
2. Open the card modal, click on the title and confirm you can see the whole text in the text area without it cut off

**Review by nk711 — APPROVED** [review]
LGTM

**Pull request merged**